coreneuron/nrnoc/nrnoc_ml.ispc

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright (c) 2016, Blue Brain Project
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+3. Neither the name of the copyright holder nor the names of its contributors
+   may be used to endorse or promote products derived from this software
+   without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/// list of data structures required for ISPC backend
+
+typedef int Datum;
+
+
+struct ThreadDatum {
+    int i;
+    double* uniform pval;
+    void* uniform _pvoid;
+};
+
+
+struct NetReceiveBuffer_t {
+    int* uniform _displ;
+    int* uniform _nrb_index;
+
+    int* uniform _pnt_index;
+    int* uniform _weight_index;
+    double* uniform _nrb_t;
+    double* uniform _nrb_flag;
+    int _cnt;
+    int _displ_cnt;
+    int _size;
+    int _pnt_offset;
+};
+
+
+struct NetSendBuffer_t {
+  int* _sendtype;  // net_send, net_event, net_move
+  int* _vdata_index;
+  int* _pnt_index;
+  int* _weight_index;
+  double* _nsb_t;
+  double* _nsb_flag;
+  uniform int _cnt;
+  uniform int _size;       /* capacity */
+  uniform int reallocated; /* if buffer resized/reallocated, needs to be copy to cpu */
+};
+
+
+struct Memb_list {
```

Comment:
some structs are suffixed with `_t`, some aren't. Is there a reason?

Reply:
they mimic original names in coreneuron...will change this completely soon..